### PR TITLE
Krnlmon support for linux networking version 2

### PR DESCRIPTION
**Pull request by n-sandeep**
Support to add bridge_id key for FDB tables
Remove other p4 tables which are not used in this LNW version

### CODE REVIEW

File: switchapi/es2k/switch_fdb.c [hunk 1]
```diff
@@ -291,9 +291,19 @@ switch_status_t switch_api_l2_forward_create(const switch_device_t device,
 
       status = switch_rif_get(device, api_l2_info->rif_handle, &rif_info);
       CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+      /* Get appropiate port_id if not already set */
+      switch_pd_to_get_port_id(&rif_info->api_rif_info);
```
Comment (ffoulkes) [inline switchapi/es2k/switch_fdb.c:296]: `switch_pd_to_get_port_id()` has two failure cases. Shouldn't it return a status code?

File: switchapi/es2k/switch_pd_utils.c [hunk 2]
```diff
@@ -194,6 +187,187 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
...
+  tdi_table_key_hdl* key_hdl = NULL;
+  tdi_table_data_hdl* data_hdl = NULL;
+  const tdi_table_hdl* table_hdl = NULL;
+  const tdi_table_info_hdl* table_info_hdl = NULL;
+  uint32_t network_byte_order = 0;
+  uint64_t get_bridge_id = 0;
+
+  krnlmon_log_debug("Entered: %s", __func__);
+
+  status = tdi_info_get(dev_id, PROGRAM_NAME, &info_hdl);
```
Comment (ffoulkes) [inline switchapi/es2k/switch_pd_utils.c:215]: These functions return the pointer to a handle, rather than the handle itself?

**Review by ffoulkes — APPROVED** [review]
LGTM
I posed one question and suggested that a function should return error status.

File: switchapi/es2k/switch_fdb.c [hunk 3]
```diff
@@ -291,9 +291,19 @@ switch_status_t switch_api_l2_forward_create(const switch_device_t device,
...
+
+      /* Get physical port ID from its corresponding PR VSI ID */
       status = switch_pd_get_physical_port_id(
           device, rif_info->api_rif_info.port_id, &api_l2_info->port_id);
       CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+      /* Get bridge ID from its physical port ID */
+      status = switch_pd_get_bridge_id(device, api_l2_info->port_id,
+                                       &api_l2_info->bridge_id);
+      CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_fdb.c:306]: What happens if the port is not part of any bridge?

File: switchapi/es2k/switch_pd_utils.c [hunk 4]
```diff
@@ -194,6 +187,187 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
...
+  status = tdi_data_field_id_with_action_get(
+      table_info_hdl,
+      LNW_SOURCE_PORT_TO_BRIDGE_MAP_TABLE_ACTION_PARAM_BRIDGE_ID, action_id,
+      &data_field_id);
+  if (status != TDI_SUCCESS) {
+    krnlmon_log_error("Unable to get action allocator ID for: %s, error: %d",
+                      ACTION_L2_FWD_AND_BYPASS_BRIDGE_PARAM_PORT, status);
+    goto dealloc_resources;
+  }
+  //    status = tdi_data_field_get_value_ptr(data_hdl, data_field_id,
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_utils.c:356]: Can we remove this instead of commenting?

File: switchapi/es2k/switch_pd_utils.c [hunk 5]
```diff
@@ -194,6 +187,187 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
...
+  tdi_target_hdl* target_hdl = NULL;
+  const tdi_device_hdl* dev_hdl = NULL;
+  tdi_session_hdl* session = NULL;
+  const tdi_info_hdl* info_hdl = NULL;
+  tdi_table_key_hdl* key_hdl = NULL;
+  tdi_table_data_hdl* data_hdl = NULL;
+  const tdi_table_hdl* table_hdl = NULL;
+  const tdi_table_info_hdl* table_info_hdl = NULL;
+  uint32_t network_byte_order = 0;
+  uint64_t get_bridge_id = 0;
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_utils.c:211]: bridge_id is uint8_t. Do we really need uint64_t here?

**Review by nupuruttarwar — COMMENTED** [review]
Can we add more information to commit message descibinmg how lnw-version 2 is different than lnw-ver1? Also, please specify that with new version, IPv6 support is removed

File: switchapi/es2k/switch_fdb.c (hunk 3 above)
Comment (n-sandeep) [reply switchapi/es2k/switch_fdb.c:306]: It is an error case, we always need to associate a port to a bridge in LNW-v2

File: switchapi/es2k/switch_pd_utils.c (hunk 5 above)
Comment (n-sandeep) [reply switchapi/es2k/switch_pd_utils.c:211]: tdi_data_field_get_value is taking uint64 as an argument which return the bridge value.
I can take it up as backlog and UT the changes to check if "tdi_data_field_get_value_ptr" can be used .

File: switchapi/es2k/switch_fdb.c (hunk 1 above)
Comment (n-sandeep) [reply switchapi/es2k/switch_fdb.c:296]: in switch_pd_to_get_port_id(), error when returning "rmac info" not found or "rmac_entry" not found if fine. There are cases where valid rmac wouldnt have been created when we call this API.

File: switchlink/switchlink_link.c [hunk 6]
```diff
@@ -88,6 +93,51 @@ static switchlink_link_type_t get_link_type(const char* info_kind) {
...
+  ifr.ifr_data = (void*)&drv;
+
+  r = ioctl(fd, SIOCETHTOOL, &ifr);
+  if (r) {
+    goto end;
+  }
+
+  strncpy(drvname, drv.driver, sizeof(drvname));
+
+  if (!memcmp(drvname, "openvswitch", strlen(drvname)) ||
```
Comment (vsureshkumarp) [inline switchlink/switchlink_link.c:130]: As part of improvements, create new macros for these driver names and utilize them when comparing drivers.

File: switchlink/switchlink_link.c [hunk 7]
```diff
@@ -88,6 +93,51 @@ static switchlink_link_type_t get_link_type(const char* info_kind) {
...
+    return false;
+  }
+
+  drv.cmd = ETHTOOL_GDRVINFO;
+  strncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));
+  ifr.ifr_data = (void*)&drv;
+
+  r = ioctl(fd, SIOCETHTOOL, &ifr);
+  if (r) {
+    goto end;
```
Comment (vsureshkumarp) [inline switchlink/switchlink_link.c:125]: It is always beneficial for debugging to log the error details at this point.  This applies to every instance in this function where error messages aren't logged before returning.

File: switchapi/es2k/switch_pd_utils.c [hunk 8]
```diff
@@ -194,6 +187,184 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
...
+
+  krnlmon_log_debug("Entered: %s", __func__);
```
Comment (vsureshkumarp) [inline switchapi/es2k/switch_pd_utils.c:213]: Remove this log.

**Review by vsureshkumarp — APPROVED** [review]
LGTM

**Comment by n-sandeep** [comment]
> * I posed one question and suggested that a function should return error status.
> * Have you done a test build to make sure dpdk isn't broken?
> * Also, you'll want to run clang-format on any source files you're committing.
>   `clang-format -i <file>...` will update them in place.

DPDK build went fine, addressed your comments and also ran clang-format

**Pull request merged**